enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -0,0 +1,493 @@
+make adding operators (via OLM) simpler on an STS enabled cluster.
+
+### User Stories
+
+* As a cluster admin I know which OLM Operators support tokenized authentication for my cloud.
+* As a cluster admin of a cluster using tokenized cloud auth I know what's required to install and upgrade OLM Operators whenever those operators manage resources that authenticate against my cloud.
+* As an Operator developer, I have a standard framework to define tokenized authentication requirements and consume them, per supported cloud.
+* As an Operator Hub browser I know which operators support tokenized cloud auth and on which clouds.
+* As the Hypershift team, where the only credential mode for clusters/customers is STS (on AWS) , the Red Hat branded
+  Operators that must reach the AWS API, should be enabled to work with STS credentials in a consistent, and automated
+  fashion that allows customer to use those operators as easily as possible, driving the use of layered products.
+
+### Goals
+
+Allow Day-2 Operators to access cloud provider resources as seamlessly as possible when being installed, used and
+updated on STS enabled clusters.
+
+While providing the above goal, allow for multi-tenancy. In this sense multi-tenancy means that an operator may
+provide different cloud provider credentials per operand such that operator users may uniquely access non-shared cloud
+resources.
+
+Operator authors have a way to notify, guide, and assist Day-2 Operator admins in providing the required cloud provider
+credentials matched to their permission needs for install and update.
+
+Ideally, a solution here will work in both HyperShift (STS always) and non-HyperShift clusters.
+
+### Non-Goals
+
+Day 1 operators are not included in this proposal, they are pre-provisioned for STS enabled clusters (at least 
+for HyperShift)
+
+
+
+## Proposal
+
+**Cloud Credential Operator (CCO) changes**: Adds a token-aware mode, while nominally set in "Manual" mode, CCO will
+look for and process CredentialRequests referenced in Operator CRs when:
+
+```bash
+# infrastructure platform is AWS
+$ oc get infrastructure cluster -o jsonpath --template '{ .status.platform }'
+AWS
+
+# credentialsMode is "Manual"
+$ oc get cloudcredential cluster -o jsonpath --template '{ .spec.credentialsMode }'
+Manual
+
+# serviceAccountIssuer is non empty
+$ oc get authentication cluster -o jsonpath --template='{ .spec.serviceAccountIssuer }'
+abutcher-oidc.s3.us-east-1.amazonaws.com
+```
+
+``. This will operate in one of two ways depending on which other
+parts of this EP for other components are adopted: specifically HyperShift/Rosa adding CCO with these token-aware
+changes or continuing with the current support for the pod identity webhook.
+
+CCO's token-aware mode will work by adding a "role-arn"-type field on operator added CredentialsRequest objects.
+This is a new API field on the CredentialsRequest spec. When CCO acquires a CredentialsRequest it will create a Secret 
```

Comment:
What's the motivation for this being on the spec? Would you not expect this to be on the status? Does a user need to be able to adjust this? What would happen if a user changed this value?

Reply:
if the ARN is changed i'd expect the secret to be updated w/ the new ARN, and consumers of the secret to pick up the change.

If it was on status i'm not clear where the ARN value would actually come from?

Reply:
I'm not sure, you'll have to educate me a bit. I would think that status is reserved for status items coming back from the cluster whereas spec (and possibly we'd put this is a subfield) is something configured by some sort of user or app process before application to the cluster? The CredentialsRequest is meant to be created by the operator, or an install process when the operator is installed. It could be updated by an admin I'd think or an automated process. @abutcher thoughts?

Reply:
> Does a user need to be able to adjust this?

User may want to be able to change the role ARN in the event that the role were deleted by some external process and the role needed to be recreated.

> What would happen if a user changed this value?

Agree with above, if the value changed the secret should be updated with the new value and consumers notice the change.

Reply:
+1 on it being in the spec. It's a desired configuration. @abutcher Does CCO add ownerRefs to the secret created from the CredReq?

Reply:
CCO does not add an `ownerRef` for secrets created for mint mode but it looks like CCO does add an annotation pointing back to the `CredentialsRequest`, 

```
metadata:
  annotations:
    cloudcredential.openshift.io/credentials-request: openshift-cloud-credential-operator/openshift-machine-api-aws
```

Reply:
going to make this resolved.